File: 30-days-of-code/day-14-scope/solution.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference
{
  private:
    vector<int> elements;

  public:
    int maximumDifference;

    Difference(vector<int> elements)
    {
        this->elements = elements;
    }
    void computeDifference()
    {
        int min = this->elements[0], max = this->elements[0];
        int length = this->elements.size();
        int i;
        for (i = 1; i < length; i++)
        {
            if (min > this->elements[i])
                min = this->elements[i];
            if (max < this->elements[i])
                max = this->elements[i];
        }
        this->maximumDifference = max - min;
    }
};

int main()
{
    int N;
    cin >> N;

    vector<int> a;

    for (int i = 0; i < N; i++)
    {
        int e;
        cin >> e;

        a.push_back(e);
    }

    Difference d(a);

    d.computeDifference();

    cout << d.maximumDifference;

    return 0;
};

File: 30-days-of-code/day-14-scope/solution.js
'use strict';

var _input = '';
var _index = 0;
process.stdin.on('data', (data) => { _input += data; });
process.stdin.on('end', () => {
  _input = _input.split(new RegExp('\n'));
  main();
});
function readLine() { return _input[_index++]; }

/**** Ignore above this line. ****/

class Difference {
  constructor(elements) {
    this._elements = elements;
    this.maximumDifference = 0;
  }

  computeDifference() {
    const min = Math.min(...this._elements);
    const max = Math.max(...this._elements);
    this.maximumDifference = max - min;
  }
}

function main() {

  let i, item, N;
  let elements = [];

  N = parseInt(readLine().trim(), 10);
  for (i = 0; i < N; i++) {
    item = parseInt(readLine().trim(), 10);
    elements.push(item);
  }
  const d = new Difference(elements);
  d.computeDifference();

  console.log(d.maximumDifference + '\n');
}
